packages/nuxt/src/components/runtime/client-delayed-component.ts

```diff
@@ -0,0 +1,175 @@
+import { defineAsyncComponent, defineComponent, h, hydrateOnIdle, hydrateOnInteraction, hydrateOnMediaQuery, hydrateOnVisible, mergeProps, watch } from 'vue'
+import type { AsyncComponentLoader, HydrationStrategy } from 'vue'
+
+/* @__NO_SIDE_EFFECTS__ */
+export const createLazyIOComponent = (loader: AsyncComponentLoader) => {
+  return defineComponent({
+    inheritAttrs: false,
+    props: {
+      hydrate: {
+        type: Object,
+        required: false,
+      },
+    },
+    emits: ['hydrated'],
+    setup (props, { attrs, emit }) {
+      const hydrated = () => { emit('hydrated') }
+      const comp = defineAsyncComponent({ loader, hydrate: hydrateOnVisible(props.hydrate as IntersectionObserverInit | undefined) })
+      // TODO: fix hydration mismatches on Vue's side. The data-allow-mismatch is ideally a temporary solution due to Vue's SSR limitation with hydrated content.
+      return () => h(comp, mergeProps(attrs, { 'data-allow-mismatch': '', 'onVnodeMounted': hydrated }))
+    },
+  })
+}
+
+/* @__NO_SIDE_EFFECTS__ */
+export const createLazyNetworkComponent = (loader: AsyncComponentLoader) => {
+  return defineComponent({
+    inheritAttrs: false,
+    props: {
+      hydrate: {
+        type: Number,
+        required: false,
+      },
+    },
+    emits: ['hydrated'],
+    setup (props, { attrs, emit }) {
+      const hydrated = () => { emit('hydrated') }
+      if (props.hydrate === 0) {
+        const comp = defineAsyncComponent(loader)
+        return () => h(comp, mergeProps(attrs, { 'data-allow-mismatch': '', 'onVnodeMounted': hydrated }))
+      }
+      const comp = defineAsyncComponent({ loader, hydrate: hydrateOnIdle(props.hydrate) })
+      // TODO: fix hydration mismatches on Vue's side. The data-allow-mismatch is ideally a temporary solution due to Vue's SSR limitation with hydrated content.
+      return () => h(comp, mergeProps(attrs, { 'data-allow-mismatch': '', 'onVnodeMounted': hydrated }))
+    },
+  })
+}
+
+/* @__NO_SIDE_EFFECTS__ */
+export const createLazyEventComponent = (loader: AsyncComponentLoader) => {
+  return defineComponent({
+    inheritAttrs: false,
+    props: {
+      hydrate: {
+        type: [String, Array],
+        required: false,
+        default: 'mouseover',
+      },
+    },
+    emits: ['hydrated'],
+    setup (props, { attrs, emit }) {
+      const hydrated = () => { emit('hydrated') }
+      // @ts-expect-error Cannot type HTMLElementEventMap in props
```

Comment:
That's because HTMLElementEventMap is not a type, it's an object

You need to use `keyof HTMLElementEventMap` but in this case I would recommend to use for all the types this:

```js
        type: null as PropType<Arguments<typeof hydrateOnInteraction>[0]>,
```

So you get the type directly from the hydration methods of vue